pkg/filters/data.go

```diff
@@ -37,6 +183,12 @@ func (af *DataFilter) Filter(eventID events.ID, data []trace.Argument) bool {
 		return true
 	}
 
+	// No need to filter the following event IDs as they have already
+	// been filtered in the kernel
+	if af.kernelDataFilter.IsKernelFilterEnabled(eventID, "pathname") {
```

Comment:
As this is called by the pipeline (hotpath), we should think in a way of check it against integers rather strings.

Reply:
Good point... I'll add a comment to revisit it in the future.